rct229/rulesets/ashrae9012019/section5/section5rule40.py

```diff
@@ -0,0 +1,238 @@
+                            USER=False, BASELINE_0=True, PROPOSED=False
+                        ),
+                        required_fields={
+                            "$": ["subclassification", "u_factor"],
+                        },
+                    )
+
+                def get_calc_vals(self, context, data=None):
+                    subsurface_b = context.BASELINE_0
+                    scc_dict_b = data["scc_dict_b"]
+                    climate_zone = data["climate_zone"]
+                    u_factor_b = subsurface_b["u_factor"]
+                    subsurface_class_b = subsurface_b["subclassification"]
+                    target_u_factor_b = ZERO.U_FACTOR
+
+                    target_u_factor_nonres_b = (
+                        table_G34_lookup(
+                            climate_zone,
+                            SCC.EXTERIOR_NON_RESIDENTIAL,
+                            DOOR,
+                            classification=subsurface_class_b,
+                        )["u_value"]
+                        if subsurface_class_b not in UNEXPECTED_DOOR
+                        else ZERO.U_FACTOR
+                    )
+
+                    target_u_factor_res_b = (
+                        table_G34_lookup(
+                            climate_zone,
+                            SCC.EXTERIOR_RESIDENTIAL,
+                            DOOR,
+                            classification=subsurface_class_b,
+                        )["u_value"]
+                        if subsurface_class_b not in UNEXPECTED_DOOR
+                        else ZERO.U_FACTOR
+                    )
+
+                    target_u_factor_semiheated_b = (
+                        table_G34_lookup(
+                            climate_zone,
+                            SCC.SEMI_EXTERIOR,
+                            DOOR,
+                            classification=subsurface_class_b,
+                        )["u_value"]
+                        if subsurface_class_b not in UNEXPECTED_DOOR
+                        else ZERO.U_FACTOR
+                    )
+
+                    if scc_dict_b == SCC.EXTERIOR_NON_RESIDENTIAL:
+                        target_u_factor_b = target_u_factor_nonres_b
+                    elif scc_dict_b == SCC.EXTERIOR_RESIDENTIAL:
+                        target_u_factor_b = target_u_factor_res_b
+                    elif scc_dict_b == SCC.SEMI_EXTERIOR:
+                        target_u_factor_b = target_u_factor_semiheated_b
+                    else:
+                        assert scc_dict_b == SCC.EXTERIOR_MIXED, (
+                            f"Surface conditioning category is not one of the five types: EXTERIOR_NON_RESIDENTIAL, "
+                            f"EXTERIOR_RESIDENTIAL, SEMI_EXTERIOR, EXTERIOR_MIXED, UNREGULATED, "
+                            f"got {scc_dict_b} instead "
+                        )
+                        if target_u_factor_nonres_b == target_u_factor_res_b:
```

Comment:
Should this if else be a separate statement instead of under the 
```python
else:
                        assert scc_dict_b == SCC.EXTERIOR_MIXED, (
                            f"Surface conditioning category is not one of the five types: EXTERIOR_NON_RESIDENTIAL, "
                            f"EXTERIOR_RESIDENTIAL, SEMI_EXTERIOR, EXTERIOR_MIXED, UNREGULATED, "
                            f"got {scc_dict_b} instead "
                        )
```

Reply:
Good point! Addressed.